From: Miguel Vadillo <vadillo@ti.com>
Date: Tue, 12 Apr 2011 23:26:26 +0000 (-0500)
Subject: syslink: ipu_pm: call complete_all in all cases
X-Git-Tag: android-2.6.35-2.3-omap4.13.1-Beta~45
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=3fe4470dd9f3334673582dda310ef46a45312260

syslink: ipu_pm: call complete_all in all cases

In case of an error while releasing resources
previously the completion of recovery was not
posted, this is not ok in all the cases since
maybe the RAT is holding an already released
resource or is corrupted.

In any case notify of a possible issue while
releasing the resources happened but allow the
reloading of the images to proceed.

Change-Id: I42e060fbb5a27a317f91b221d65537f637432d6f
Reported-by: Juan Gutierrez <jgutierrez@ti.com>
Signed-off-by: Miguel Vadillo <vadillo@ti.com>

Index: kernel/drivers/dsp/syslink/ipu_pm/ipu_pm.c
===================================================================
--- kernel.orig/drivers/dsp/syslink/ipu_pm/ipu_pm.c	2011-06-17 18:25:37.000000000 +0900
+++ kernel/drivers/dsp/syslink/ipu_pm/ipu_pm.c	2011-06-19 22:29:19.000000000 +0900
@@ -566,11 +566,13 @@
 	}
 
 	global_rcb->rat = used_res_mask;
-	if (!used_res_mask)
-		goto complete_exit;
 
-	pr_warning("%s:Not all resources were released", __func__);
-	return;
+	/* Maybe an already released resource or currupted RAT
+	 * anyway call complete to allow the reload of the images
+	 */
+	if (used_res_mask)
+		pr_warning("%s: Not all resources were released\n", __func__);
+
 complete_exit:
 	complete_all(&ipu_clean_up_comp);
 	return;
